Remove pmm-sumo from codeowners

pmm-sumo opened the pull request:
**Description:** 

Remove @pmm-sumo from the list of codeowners as I'm no longer able to contribute enough time for OpenTelemetry Community (#12493)

This is also a followup from #12596 and a [suggestion](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12596#issuecomment-1192589716) to break it into separate PR's for each component

pmm-sumo committed:
Remove pmm-sumo from codeowners and sponsors

pmm-sumo commented:
(BTW, I'm no longer to apply "Skip Changelog" label and I think it would fit here)

TylerHelmuth commented on .github/CODEOWNERS:
```diff
@@ -64,10 +64,10 @@ exporter/sentryexporter/                             @open-telemetry/collector-c
 exporter/skywalkingexporter/                         @open-telemetry/collector-contrib-approvers @liqiangz
 exporter/splunkhecexporter/                          @open-telemetry/collector-contrib-approvers @atoulme @dmitryax
-exporter/sumologicexporter/                          @open-telemetry/collector-contrib-approvers @pmm-sumo @sumo-drosiek
+exporter/sumologicexporter/                          @open-telemetry/collector-contrib-approvers @swiatekm-sumo @sumo-drosiek
```
I support removing @pmm-sumo, but we've decided that all [code owners going forward need to be members of the OpenTelemetry organization](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner), and there was not an exception made for vender-specific components.  Is there anyone else from Sumo that is already a member that could be added?

pmm-sumo replied:
@swiatekm-sumo considering that you [already made several contributions](https://github.com/open-telemetry/opentelemetry-collector-contrib/pulls?q=is%3Apr+author%3Aswiatekm-sumo+is%3Aclosed) I would suggest to apply for membership as the preferred solution :)

pmm-sumo replied:
@TylerHelmuth I haven't seen followup on my suggestion, so to avoid blocking this PR further, I removed this entry. I believe this was the only outstanding item

pmm-sumo committed:
Update sumologicexporter CODEOWNERS

Excluding @swiatekm-sumo from the proposal as the he is not (yet) member of OpenTelemetry org

pmm-sumo committed:
Merge branch 'main' into remove-from-codeowners

TylerHelmuth reviewed (approved):
Thanks again for all your contributions!

pmm-sumo commented:
This is probably a simple change, wanted to bring it up again to avoid unnecessary confusion when assigning issues/etc. @open-telemetry/collector-contrib-maintainer :)

djaglowski committed:
Merge branch 'main' into remove-from-codeowners

djaglowski reviewed (approved):
It's been great working with you, @pmm-sumo. Best of luck!

djaglowski committed:
Merge branch 'main' into remove-from-codeowners

pmm-sumo commented:
Thank you @djaglowski! Same here :)
